config/auth_m2m.go

```diff
@@ -55,26 +54,11 @@ func oidcEndpoints(ctx context.Context, cfg *Config) (*oauthAuthorizationServer,
 		}, nil
 	}
 	oidc := fmt.Sprintf("%s/oidc/.well-known/oauth-authorization-server", prefix)
-	oidcResponse, err := http.Get(oidc)
-	if err != nil {
-		return nil, fmt.Errorf("fetch .well-known: %w", err)
-	}
-	if oidcResponse.StatusCode != 200 {
-		logger.Warnf(ctx, "Failure getting OIDC response. Response status: %s", oidcResponse.Status)
-		return nil, errOAuthNotSupported
-	}
-	if oidcResponse.Body == nil {
-		return nil, fmt.Errorf("fetch .well-known: empty body")
-	}
-	defer oidcResponse.Body.Close()
-	raw, err := io.ReadAll(oidcResponse.Body)
-	if err != nil {
-		return nil, fmt.Errorf("read .well-known: %w", err)
-	}
 	var oauthEndpoints oauthAuthorizationServer
-	err = json.Unmarshal(raw, &oauthEndpoints)
+	err := cfg.refreshClient.Do(ctx, "GET", oidc,
+		httpclient.WithResponseUnmarshal(&oauthEndpoints))
 	if err != nil {
-		return nil, fmt.Errorf("parse .well-known: %w", err)
+		return nil, errOAuthNotSupported
```

Comment:
Just want to check, is there a reason we want to be less descriptive on type of error message we show to user? 

Before we use to show empty body / response status / err  but now we will only show `databricks OAuth is not supported for this host`.

Reply:
While `parse .well-known:` doesn't make sense to be thrown on user end. It would be good to throw the `err` along with error message that OAuth is not supported. Stamping to unblock.